File: src/main/java/net/hellomouse/xeno_early_start/registries/ProgressionModEntityRegistry.java
package net.hellomouse.xeno_early_start.registries;

import net.hellomouse.xeno_early_start.ProgressionMod;
import net.hellomouse.xeno_early_start.entity.BrickEntity;
import net.minecraft.entity.EntityType;
import net.minecraft.entity.SpawnGroup;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;
import net.minecraftforge.registries.RegistryObject;

public class ProgressionModEntityRegistry {
    public static final DeferredRegister<EntityType<?>> DEF_REG = DeferredRegister.create(ForgeRegistries.ENTITY_TYPES, ProgressionMod.MODID);
    public static final RegistryObject<EntityType<BrickEntity>> BRICK = DEF_REG.register("brick",
            () -> EntityType.Builder.create((EntityType.EntityFactory<BrickEntity>) BrickEntity::new, SpawnGroup.MISC)
                    .setDimensions(0.5F, 0.5F)
                    .build("brick"));
}

File: src/main/kotlin/net/hellomouse/xeno_early_start/registries/ProgressionModEntityRegistry.kt
package net.hellomouse.xeno_early_start.registries

import net.hellomouse.xeno_early_start.ProgressionMod
import net.hellomouse.xeno_early_start.entity.BrickEntity
import net.minecraft.entity.EntityType
import net.minecraft.entity.SpawnGroup
import net.minecraft.world.World
import net.minecraftforge.registries.DeferredRegister
import net.minecraftforge.registries.ForgeRegistries
import net.minecraftforge.registries.RegistryObject
import java.util.function.Supplier

object ProgressionModEntityRegistry {
    val DEF_REG: DeferredRegister<EntityType<*>> =
        DeferredRegister.create<EntityType<*>>(ForgeRegistries.ENTITY_TYPES, ProgressionMod.Companion.MODID)

    @JvmField
    val BRICK: RegistryObject<EntityType<BrickEntity>> = DEF_REG.register<EntityType<BrickEntity>>(
        "brick",
        Supplier {
            EntityType.Builder.create<BrickEntity>(EntityType.EntityFactory { brickEntityEntityType: EntityType<BrickEntity>, world: World ->
                BrickEntity(
                    brickEntityEntityType,
                    world
                )
            }, SpawnGroup.MISC)
                .setDimensions(0.5f, 0.5f)
                .build("brick")
        })
}
